File: java/69.java
class Solution {
    public int mySqrt(int x) {
        int left = -1;
        int right = 46341;
        while(left + 1 < right){
            int middle = (left+right)/2;
            if(middle * middle <= x) {
                left = middle;
            } else {
                right = middle;
            }
        }
        return left;
    }
}

File: py/69.py
class Solution(object):
    def mySqrt(self, x):
        """
        :type x: int
        :rtype: int
        """
        left = 0
        right = x+1
        while left + 1 < right:
            middle = (left+right)//2
            if middle * middle <= x: 
                left = middle
            else: 
                right = middle
        return left
        
